src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -2153,4 +2161,58 @@ private static bool IsXmpTextData(ReadOnlySpan<byte> keywordBytes)
 
     private void SwapScanlineBuffers()
         => (this.scanline, this.previousScanline) = (this.previousScanline, this.scanline);
+
+    // FIXME: Maybe this could be a .Mutate(x => x.ApplyIccProfile(destinationProfile)) ? Nothing related to png here
+    private static void ApplyIccProfile<TPixel>(Image<TPixel> image, IccProfile sourceProfile, IccProfile destinationProfile)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        ColorConversionOptions options = new()
+        {
+            SourceIccProfile = sourceProfile,
+            TargetIccProfile = destinationProfile,
+        };
+
+        ColorProfileConverter converter = new(options);
+
+        image.ProcessPixelRows(pixelAccessor =>
```

Comment:
I would use `ProcessPixelRowsAsVector4` passing `PixelConversionModifiers.Scale` then you get the SIMD optimized conversion.